app/Http/Controllers/Api/RestoreDatabaseController.php

```
@@ -0,0 +1,333 @@
+        // Validate PITR requirements
+        if ($request->filled('target_time')) {
+            if (! $backupConfig->enable_pitr) {
+                return response()->json([
+                    'message' => 'Point-in-time recovery is not enabled for this backup configuration.',
+                    'errors' => ['target_time' => ['PITR must be enabled to restore to a specific time.']],
+                ], 422);
+            }
+
+            // Validate timestamp format
+            try {
+                $targetTime = new \DateTime($request->target_time);
```

PHPMD (2.15.0)

Missing class import via use statement (line '169', column '35'). (undefined)

(MissingImport)